### ISSUE TITLE
--allow-net=:8080 broken

### ISSUE THREAD

**Issue by brandonkal** [denoland/deno#6685]
If any code uses shorthand notation:
```ts
import { serve } from "https://deno.land/std/http/server.ts"
const PORT = 8080
serve(`:${PORT}`)
```
then any variant of 
```
deno run --allow-net=:8080 server.ts
deno run --allow-net=localhost:8080 server.ts
deno run --allow-net=0.0.0.0:8080 server.ts
deno run --allow-net=127.0.0.1:8080 server.ts
```
fails.

Unfortunately, a lot of libraries in x use this pattern which makes locking down permissions impossible without forking all those libraries.

**Comment by ry** [comment]
Maybe this was introduced in 74c260517a40d9644960b01415824e1c6c646fde ?

I think this shouldn't be a terrible difficult or involved fix. I will mark as "Good First Issue".

**Comment by KrisChambers** [comment]
I would like to take a stab at this one.

**Comment by jacobgc** [comment]
Just tested this with deno version 1.1.2 which is before my commit (74c2605) and can confirm that this issue is still present.

**Comment by utkzas** [comment]
I also want to contribute to this project. Can someone tell me how should I start?

**Comment by jsouto18** [comment]
I've tested this on the latest release and I've managed to execute the code successfully running `
deno run --allow-net=:8080 server.ts` or `
deno run --allow-net=0.0.0.0:8080 server.ts
` which seems to be the intended behaviour as the serve function binds to `0.0.0.0:8080`.
@ry is this the intended behaviour?
The only other thing I found odd was trying to run `serve(127.0.0.1:8080)` and running `deno run --allow-net=localhost:8080 server.ts` fails because localhost is not resolved to the configured IP, for example in /etc/hosts.
